Fixes build errors with gcc 4.9

tt4g opened the pull request:
Ref bincrafters/community#637, jtv/libpqxx#161

This patch is fixes build errors with gcc 4.9, ​​not necessary for libpqxx 6.3.1 and later versions.

tt4g committed:
Fixes build errors with gcc 4.9

This patch is fixes build errors with gcc 4.9, ​​not necessary for libpqxx 6.3.1 and later versions.

tt4g commented:
I noticed.

This project uses its own `CMakeLists.txt`.

I guess this is because libpqxx 6.2.4 did not support CMake, but libpqxx 6.3.1 support CMake build.

Is there policy on CMake build of libpxx 6.3.1?

SSE4 commented:
@tt4g if project itself supports CMake, then use it

tt4g commented:
> @tt4g if project itself supports CMake, then use it

I'll try using project CMake.

It may take time.

tt4g commented:
Another problem is that the AppleClang compiler on XCode 7.3 has encountered an error.

https://travis-ci.org/bincrafters/conan-libpqxx/jobs/499104663#L4250
```bash
$ strconv.cxx:231:7: error: thread-local storage is not supported for the current target
      thread_local dumb_stringstream<T> S;
```

It seems to AppleClang with  Xcode 7.3 does not support this C++ Features: https://stackoverflow.com/questions/33358417/how-to-get-support-for-thread-local-on-mac-osx-clang

@SSE4 Do I need to deal with this problem?

SSE4 commented:
@tt4g nope, just remove Xcode 7,3 from `travis.yml`. optionally, add check into `configure` method to throw `ConanInvalidConfiguration` exception for old Xcode versions.

tt4g committed:
Remove support for AppleClang earlier to 8.0

libpqxx requires C++ Thread-local storage features.
XCode earlier to 8.0 are not able to build libpqxx.

tt4g commented:
It seems it will still take time to change to use libpqxx's CMake,
So, I would like to make a new PR.

@SSE4 Would you merge this PR?

tt4g commented:
Thanks.
